### PR TITLE
test(auth): add integration tests for regional access boundaries

### PR DESCRIPTION
**Pull request by alvarowolfx**
Towards #4186 #5719

### CODE REVIEW

File: tests/integration-auth/src/lib.rs [hunk 1]
```diff
@@ -664,3 +705,37 @@ pub async fn id_token_impersonated() -> anyhow::Result<()> {
...
+        let headers = match cached_headers {
+            CacheableResource::New { data, .. } => data,
+            CacheableResource::NotModified => {
+                unreachable!("should always get new headers when Extensions is empty")
+            }
+        };
+        let locations = headers.get("x-goog-allowed-locations");
+        if locations.is_some() {
+            locations_found = true;
+            break;
```
Comment (dbolduc) [inline tests/integration-auth/src/lib.rs]: optional nit: could `return Ok(());`

File: tests/integration-auth/src/lib.rs [hunk 2]
```diff
@@ -664,3 +705,37 @@ pub async fn id_token_impersonated() -> anyhow::Result<()> {
...
+        }
+        tokio::time::sleep(tokio::time::Duration::from_millis(200)).await;
+    }
+
+    assert!(
+        locations_found,
+        "x-goog-allowed-locations header was not injected into the headers"
+    );
+
+    Ok(())
```
Comment (dbolduc) [inline tests/integration-auth/src/lib.rs]: ... and `panic!` / `anyhow::bail!` here

**Review by dbolduc — APPROVED** [review]
